merino/providers/manifest/provider.py

```diff
@@ -86,3 +94,23 @@ def _should_fetch(self) -> bool:
     def get_manifest_data(self) -> ManifestData | None:
         """Return manifest data"""
         return self.manifest_data
+
+    def get_icon_url(self, url: str | HttpUrl) -> str | None:
+        """Get icon URL for a domain.
+
+        Args:
+            url: Full URL to extract domain from (string or HttpUrl)
+
+        Returns:
+            Icon URL if found, None otherwise
+        """
+        try:
+            url_str = str(url)
+            domain = urlparse(url_str).netloc.replace("www.", "")
+            idx = self.lookup_table.get(domain)
+            if idx is not None and self.manifest_data is not None:
+                return self.manifest_data.domains[idx].icon
+
```

Comment:
```suggestion
idx = self.lookup_table.get(domain, -1)  # Default to -1 if domain is not found
if idx >= 0 and self.manifest_data is not None:
    return self.manifest_data.domains[idx].icon

```
this might be better since idx is an int